src/main/java/com/vonhof/webi/WebiContext.java

```diff
@@ -64,7 +65,14 @@ protected WebiContext(String base,String path, Request jettyRequest, HttpServlet
         this.request = request;
         this.response = response;
         httpMethod = HttpMethod.valueOf(request.getMethod());
-        parmMap = new ParmMap(request.getParameterMap());
```

Comment:
I have checked a couple of different Jetty implementations (the one used by Webi) of `ServletRequest#getParameterMap` and I don't see anything about the things you mention in the Jira issue, ie. that the method 'reads the body (input stream) if it is a POST request with content type “application/x-www-form-urlencoded”' (I assume you mean swallows the body).

Can you point me to the correct implementation of `ServletRequest#getParameterMap` that works like this so I can understand why your fix works?

Reply:
You’re right it is largely undocumented - I identified it by stepping through code and see when the body was consumed